### ISSUE TITLE
Cannot connect with rabbitmqclusters that's using Vault as secret backend

### ISSUE THREAD

**Issue by ChunyiLyu** [rabbitmq/messaging-topology-operator#364]
## Describe the bug

See bug created in cluster operator: https://github.com/rabbitmq/cluster-operator/issues/1030

When creating topology objects in Vault enabled rabbitmqcluster, due to the incomplete `.status`, messaging topology operator panic with error

```
{"level":"info","ts":1652195700.6349993,"msg":"Initiating lifecycle management of Vault token"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x14c9526]

goroutine 519 [running]:
github.com/rabbitmq/messaging-topology-operator/internal.ParseRabbitmqClusterReference({0x1afb5e8, 0xc000ae6120}, {0x1b2bbd8, 0xc000532be0}, {{0xc0001334a0, 0x12}, {0x0, 0x0}, 0x0}, {0xc00090c1f0, ...}, ...)
	/workspace/internal/cluster_reference.go:97 +0x4c6
github.com/rabbitmq/messaging-topology-operator/controllers.(*QueueReconciler).Reconcile(0xc0003283c0, {0x1afb5e8, 0xc000ae6120}, {{{0xc00090c1f0, 0x17862a0}, {0xc00090c1e0, 0x30}}})
	/workspace/controllers/queue_controller.go:59 +0x25c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc000534000, {0x1afb5e8, 0xc000ae6060}, {{{0xc00090c1f0, 0x17862a0}, {0xc00090c1e0, 0x413a94}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:114 +0x26f
```

**Comment by ChunyiLyu** [comment]
[PR that sets service reference in rabbitmqcluster](https://github.com/rabbitmq/cluster-operator/pull/1031). Will need to wait for a new version of cluster operator is cut.

**Comment by ChunyiLyu** [comment]
Fixes are released in the latest released operators: cluster operator `1.13.0` and topology operator `1.6.0`. Closing
